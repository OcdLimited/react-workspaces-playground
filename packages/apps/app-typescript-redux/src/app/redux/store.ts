import { configureStore, ThunkAction, Action, StoreEnhancer } from '@reduxjs/toolkit';
import withReduxEnhancer from 'addon-redux/enhancer';
import createSagaMiddleware from 'redux-saga';
import rootReducer from './rootReducer';
import rootSaga from './sagas/sagas';

const enhancers = new Array<StoreEnhancer>();

if (process.env.NODE_ENV !== 'production') {
	enhancers.push(withReduxEnhancer);
}

const sagaMiddleware = createSagaMiddleware();

export const store = {
	...configureStore({
		reducer: rootReducer,
		enhancers: enhancers,
		middleware: [sagaMiddleware],
	}),
	injectedSagas: [],
	runSaga: sagaMiddleware.run,
};

sagaMiddleware.run(rootSaga);

if (process.env.NODE_ENV === 'development' && module.hot) {
	module.hot.accept('./rootReducer', () => {
		const newRootReducer = require('./rootReducer').default;
		store.replaceReducer(newRootReducer);
	});
}

export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
export type AppDispatch = typeof store.dispatch;

export default store;
