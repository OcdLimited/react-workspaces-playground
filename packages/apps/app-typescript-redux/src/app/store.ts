import { configureStore, ThunkAction, Action, StoreEnhancer } from '@reduxjs/toolkit';
import withReduxEnhancer from 'addon-redux/enhancer';
import counterReducer from '../features/counter/counterSlice';

const enhancers = new Array<StoreEnhancer>();

if (process.env.NODE_ENV !== 'production') {
  enhancers.push(withReduxEnhancer);
}

export const store = configureStore({
  reducer: {
    counter: counterReducer,
  },
  enhancers: enhancers,
});

export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
