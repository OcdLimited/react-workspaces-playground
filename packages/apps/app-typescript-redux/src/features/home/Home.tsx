import React from 'react';
import clsx from 'clsx';
import { makeStyles, Theme } from '@material-ui/core/styles';
import { Chart } from '@ocdlimited/abp.react.theme.shared';
import { Grid, Paper } from '@material-ui/core';
import Deposits from './Deposits';
import Orders from './Orders';

const useStyles = makeStyles((theme: Theme) => ({
	paper: {
		padding: theme.spacing(2),
		display: 'flex',
		overflow: 'auto',
		flexDirection: 'column',
	},
	fixedHeight: {
		height: 240,
	},
}));

export function Home() {
	const classes = useStyles();

	const fixedHeightPaper = clsx(classes.paper, classes.fixedHeight);

	return (
		<React.Fragment>
			<Grid container spacing={3}>
				{/* Chart */}
				<Grid item xs={12} md={8} lg={9}>
					<Paper className={fixedHeightPaper}>
						<Chart />
					</Paper>
				</Grid>
				{/* Recent Deposits */}
				<Grid item xs={12} md={4} lg={3}>
					<Paper className={fixedHeightPaper}>
						<Deposits />
					</Paper>
				</Grid>
				{/* Recent Orders */}
				<Grid item xs={12}>
					<Paper className={classes.paper}>
						<Orders />
					</Paper>
				</Grid>
			</Grid>
		</React.Fragment>
	);
}

export default Home;
