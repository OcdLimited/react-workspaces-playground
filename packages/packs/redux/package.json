{
	"name": "@ocdlimited/abp.react.redux",
	"version": "0.0.4-alpha.0",
	"main": "src/index.ts",
	"main:src": "src/index.ts",
	"scripts": {
		"test": "react-scripts test",
		"lint": "eslint ./src/**/*.ts ./src/*.ts --max-warnings=0 --format=codeframe"
	},
	"dependencies": {
		"@ocdlimited/react-scripts": "^0.0.4-alpha.0",
		"@reduxjs/toolkit": "^1.2.5",
		"@testing-library/jest-dom": "^4.2.4",
		"@testing-library/react": "^9.4.0",
		"@testing-library/user-event": "^8.0.3",
		"@types/jest": "^24.0.25",
		"@types/node": "^13.1.6",
		"@types/react": "^16.9.17",
		"@types/react-dom": "^16.9.4",
		"@types/react-redux": "^7.1.7",
		"@typescript-eslint/eslint-plugin": "^2.15.0",
		"@typescript-eslint/parser": "^2.15.0",
		"addon-redux": "^1.1.0",
		"react": "^16.13.1",
		"react-dom": "^16.13.1",
		"react-redux": "^7.2.0",
		"react-router": "6.0.0-alpha.4",
		"react-router-dom": "6.0.0-alpha.4",
		"redux-saga": "1.1.3",
		"typescript": "~3.8.2"
	},
	"eslintConfig": {
		"extends": "react-app"
	},
	"browserslist": [
		">0.2%",
		"not dead",
		"not ie <= 11",
		"not op_mini all"
	],
	"jest": {
		"collectCoverageFrom": [
			"src/**/*.{ts,tsx}",
			"!src/setupTests.{ts,tsx}",
			"!src/serviceWorker.{ts,jsx}",
			"!src/**/*.(spec|test|stories).{ts,tsx}"
		],
		"coverageThreshold": {
			"global": {
				"statements": 95,
				"branches": 80,
				"functions": 90,
				"lines": 98
			}
		}
	},
	"license": "MIT"
}
