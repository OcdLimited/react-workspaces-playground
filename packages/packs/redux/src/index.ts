export * from './sagas';
export * from './reducers';
export * from './store';
export * from './types';
