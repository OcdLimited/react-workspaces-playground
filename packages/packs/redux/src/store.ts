import { configureStore, StoreEnhancer, Store, AnyAction } from '@reduxjs/toolkit';
import withReduxEnhancer from 'addon-redux/enhancer';
import createSagaMiddleware from 'redux-saga';
import { createApiMiddleware, appConfigSaga } from '@ocdlimited/abp.react.core';
import { createRootReducer } from './reducers/rootReducer';

export interface InjectableStore extends Store<any, AnyAction> {
	injectedReducers: any;
	injectedSagas: any;
	runSaga(saga: any, args: any): any;
}

const enhancers: StoreEnhancer[] = [];

if (process.env.NODE_ENV !== 'production') {
	enhancers.push(withReduxEnhancer);
}

export function buildStore(rootReducer: any, preloadedState?: any): InjectableStore {
	const sagaMiddleware = createSagaMiddleware();

	const store = {
		...configureStore({
			reducer: createRootReducer(rootReducer),
			enhancers: enhancers,
			middleware: [createApiMiddleware(), sagaMiddleware],
			preloadedState,
		}),
		injectedSagas: [],
		injectedReducers: { ...rootReducer },
		runSaga: sagaMiddleware.run,
	};

	sagaMiddleware.run(appConfigSaga);

	/* istanbul ignore next */
	if (process.env.NODE_ENV === 'development' && module.hot) {
		if (module.hot) {
			// Enable Webpack hot module replacement for reducers
			module.hot.accept('./reducers', () => {
				const nextReducer = createRootReducer(store.injectedReducers);
				store.replaceReducer(nextReducer);
			});
		}
	}

	return store;
}
