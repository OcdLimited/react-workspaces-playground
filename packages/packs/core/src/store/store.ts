import { combineReducers } from '@reduxjs/toolkit';
import { InjectableStore, buildStore as innerBuildStore } from '@ocdlimited/abp.react.redux';
import { apiReducer, createApiMiddleware } from '../api';
import { appConfigReducer, appConfigSaga } from '../application-configuration';
import { toastReducer } from '../toast';
import { tokenReducer } from '../token';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function createRootReducer(injectedReducers: any): any {
	return combineReducers({
		api: apiReducer,
		config: appConfigReducer,
		toast: toastReducer,
		token: tokenReducer,
		...injectedReducers,
	});
}

export function buildStore(rootReducer: unknown, preloadedState?: unknown): InjectableStore {
	const reducer = createRootReducer(rootReducer);
	return innerBuildStore(reducer, createRootReducer, preloadedState, [createApiMiddleware()], [appConfigSaga]);
}

export default buildStore;
