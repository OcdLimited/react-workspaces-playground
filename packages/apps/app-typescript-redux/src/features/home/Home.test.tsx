import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../app/redux/store';
import Home from './Home';

it('renders', () => {
	const { container, getByText } = render(
		<Provider store={store}>
			<MemoryRouter>
				<Home />
			</MemoryRouter>
		</Provider>,
	);

	expect(getByText(/Today/i)).toBeInTheDocument();
	expect(container).toMatchSnapshot();
});
