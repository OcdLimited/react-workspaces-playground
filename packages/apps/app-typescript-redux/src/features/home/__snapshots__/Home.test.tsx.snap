// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-9"
    >
      <div
        class="MuiPaper-root makeStyles-paper-1 makeStyles-fixedHeight-2 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <h1
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Today
        </h1>
        <div
          class="recharts-responsive-container"
          style="width: 100%; height: 100%;"
        >
          <div
            style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
    >
      <div
        class="MuiPaper-root makeStyles-paper-1 makeStyles-fixedHeight-2 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <h1
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Recent Deposits
        </h1>
        <p
          class="MuiTypography-root MuiTypography-h4"
        >
          $3,024.00
        </p>
        <p
          class="MuiTypography-root makeStyles-depositContext-164 MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          on 15 March, 2019
        </p>
        <div>
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="#"
          >
            View balance
          </a>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <h1
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Recent Orders
        </h1>
        <table
          class="MuiTable-root"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                scope="col"
              >
                Date
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                scope="col"
              >
                Name
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                scope="col"
              >
                Ship To
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                scope="col"
              >
                Payment Method
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
                scope="col"
              >
                Sale Amount
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                16 Mar, 2019
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                Elvis Presley
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                Tupelo, MS
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                VISA ⠀•••• 3719
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
              >
                312.44
              </td>
            </tr>
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                16 Mar, 2019
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                Paul McCartney
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                London, UK
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                VISA ⠀•••• 2574
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
              >
                866.99
              </td>
            </tr>
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                16 Mar, 2019
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                Tom Scholz
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                Boston, MA
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                MC ⠀•••• 1253
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
              >
                100.81
              </td>
            </tr>
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                16 Mar, 2019
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                Michael Jackson
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                Gary, IN
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                AMEX ⠀•••• 2000
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
              >
                654.39
              </td>
            </tr>
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                15 Mar, 2019
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                Bruce Springsteen
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                Long Branch, NJ
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              >
                VISA ⠀•••• 5919
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
              >
                212.79
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="makeStyles-seeMore-171"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="#"
          >
            See more orders
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
